### PR TITLE
Fix isIP: closes #2972, move integration tests to unit tests and…

### LINKED ISSUE #2972: validate isIP fail unexpected when Model.create
The `isIP` validator failed when I give a right ip, sequelize version `2.0.0-rc7`.
I'm looking for the reason, maybe the `ture` value of `isIP` conflicting with the second parameter of `validator.isIP`, it expect 4 or 6 or empty.
But I don't know why the `Model.build` executes success, the test code is below:

Model
```javascript
module.exports = function(sequelize, DataTypes) {
    var Test = sequelize.define('Test', {
        ip: {
            type: DataTypes.STRING,
            allowNull: false,
            validate: {
                isIP: true
            }
        }
    });

    return Test;
};
```
call create and build
```javascript
var Models = require('../models');
var Test = Models.Test;

// throw error
Test.create({
    ip: '127.0.0.1'
}).catch(function(e) {
    console.log(e);
    // { [SequelizeValidationError: Validation error]
    //   name: 'SequelizeValidationError',
    //   message: 'Validation error',
    //   errors:
    //    [ { message: 'Validation isIP failed',
    //        type: 'Validation error',
    //        path: 'ip',
    //        value: 'Validation isIP failed',
    //        __raw: 'Validation isIP failed' } ] }
});

// success
Test.build({ip: '127.0.0.1'});
```

### PR DESCRIPTION
**Pull request by BridgeAR**
… introduce utils.sliceArgs helper. The last is used to prevent deoptimization while handling arguments.

### CODE REVIEW

File: lib/data-types.js [hunk 1]
```diff
@@ -531,9 +531,15 @@ VIRTUAL.prototype.key = VIRTUAL.key = 'VIRTUAL';
  */
 var ENUM = function(value) {
   var options = typeof value === 'object' && !Array.isArray(value) && value || {
-    values: Array.prototype.slice.call(arguments).reduce(function(result, element) {
-      return result.concat(Array.isArray(element) ? element : [element]);
-    }, [])
+    values: (function (args) {
+      var tmp = new Array(args.length);
```
Comment (mickhansen) [inline lib/data-types.js]: Makes the code a lot less readable for something that is only used once ever in an application.

Comment (BridgeAR) [inline lib/data-types.js]: Utils can't be used because it'd be circular but, in this case it's not really bad to keep the old way, that's true.

**Comment by janmeier** [comment]
LGTM  - needs a rebase

**Comment by BridgeAR** [comment]
Rebased

**Pull request merged**